src/cpp/include/UTIL/CheckCollections.h

```diff
@@ -0,0 +1,79 @@
+#ifndef CheckCollections_h
+#define CheckCollections_h 1
+
+#include "lcio.h"
+
+#include <string>
+#include <unordered_map>
+#include <set>
+
+namespace UTIL {
+
+  
+/** Utility class for checking and patching events with respect to collections that are not present 
+ *  in every event of a set of files. 
+ *  
+ *  \author F.Gaede, DESY
+ *  \date Dec 2022
+ */
+  class CheckCollections {
+    
+    typedef std::vector<std::pair<std::string,std::string>> Vector ;
+    
+  public: 
+    
+    /** Convenient c'tor.
+     */
+    CheckCollections() = default ;
+
+    /** d'tor */
+    virtual ~CheckCollections() = default ;
+
+    /** Checks the file for missing collections - can be called repeadedly on different files.
+     */
+    void checkFile( const std::string& fileName ) ;
+
+    /** Checks all files for missing collections.
+     */
+    void checkFiles( const std::vector<std::string>& fileNames ) ;
+
+    
+    /** dump result of check to stream */
+    void print(  std::ostream& os ) const ;
+    
+
+    /** Returns the collections that are not present in all events checked with checkFiles() with their names and types.
+     */
+    Vector getMissingCollections() const ;
+
+    /** Returns the collections that are present in all events checked with checkFiles() with their names and types.
+     */
+    Vector getConsistentCollections() const ;
+
+    /** Add a collection with (name,type) that should be added to events in patchEvent().
+     */
+    void addPatchCollection(const std::string name, std::string type){
+      _patchCols.push_back( {name, type} ) ;
+    }
+
+    /** Add a all collections as Vector(name,type), e.g. retrieved from getMissingCollections()  that should be added to events in patchEvent().
+     */
+    void addPatchCollections(Vector cols){
+      for(const auto& p : cols)
+	_patchCols.push_back( p ) ;
```

Comment:
This will now no longer work as before with the `set`, as it is now possible to add the same collections multiple times, so this would have to be something like
```suggestion
       if (const auto it = std::find(_patchCols.begin(), _patchCols.end(), p); it == _patchCols.end()) {
	  _patchCols.push_back( p ) ;
	}
```